Sources/FluentKit/Migration/Migrator.swift

```diff
@@ -129,43 +177,43 @@ public struct Migrator {
         }
     }
     
-    private func revertMigrationLog() -> EventLoopFuture<Void> {
-        MigrationLog.migration.revert(on: self.database(nil))
+    private func revertMigrationLog(on databaseID: DatabaseID?) -> EventLoopFuture<Void> {
+        MigrationLog.migration.revert(on: self.database(databaseID))
     }
     
-    private func lastBatchNumber() -> EventLoopFuture<Int> {
-        MigrationLog.query(on: self.database(nil)).sort(\.$batch, .descending).first().map { log in
+    private func lastBatchNumber(on databaseID: DatabaseID?) -> EventLoopFuture<Int> {
+        MigrationLog.query(on: self.database(databaseID)).sort(\.$batch, .descending).first().map { log in
             log?.batch ?? 0
         }
     }
     
-    private func preparedMigrations() -> EventLoopFuture<[Migrations.Item]> {
-        MigrationLog.query(on: self.database(nil)).all().map { logs -> [Migrations.Item] in
+    private func preparedMigrations(on databaseID: DatabaseID?) -> EventLoopFuture<[Migrations.Item]> {
+        MigrationLog.query(on: self.database(databaseID)).all().map { logs -> [Migrations.Item] in
             self.migrations.storage.filter { storage in
                 logs.contains { log in
                     storage.migration.name == log.name
-                }
+                } && storage.id == databaseID
             }.reversed()
         }
     }
     
-    private func preparedMigrations(batch: Int) -> EventLoopFuture<[Migrations.Item]> {
-        MigrationLog.query(on: self.database(nil)).filter(\.$batch == batch).all().map { logs in
+    private func preparedMigrations(batch: Int, on databaseID: DatabaseID?) -> EventLoopFuture<[Migrations.Item]> {
+        MigrationLog.query(on: self.database(databaseID)).filter(\.$batch == batch).all().map { logs in
             self.migrations.storage.filter { storage in
                 logs.contains { log in
                     storage.migration.name == log.name
-                }
+                } && storage.id == databaseID
             }.reversed()
         }
     }
     
-    private func unpreparedMigrations() -> EventLoopFuture<[Migrations.Item]> {
-        return MigrationLog.query(on: self.database(nil))
+    private func unpreparedMigrations(on databaseID: DatabaseID?) -> EventLoopFuture<[Migrations.Item]> {
+        return MigrationLog.query(on: self.database(databaseID))
             .all()
             .map
         { logs -> [Migrations.Item] in
             return self.migrations.storage.compactMap { item in
-                if logs.filter({ $0.name == item.migration.name }).count == 0 {
+                if item.id == databaseID && logs.filter({ $0.name == item.migration.name }).count == 0 {
```

Comment:
`logs` is an `Array`, you should be able to use `!logs.contains(where: { $0.name == item.migration.name })` instead of comparing the count like that.